platform/android/java/lib/src/org/godotengine/godot/Godot.kt

```diff
@@ -670,6 +676,48 @@ class Godot(private val context: Context) {
 		for (plugin in pluginRegistry.allPlugins) {
 			plugin.onMainActivityResult(requestCode, resultCode, data)
 		}
+
+		if (requestCode == FILE_PICKER_REQUEST) {
+			if (resultCode == Activity.RESULT_CANCELED) {
+				Log.d(TAG, "File picker canceled")
+				GodotLib.filePickerCallback(false, emptyArray(), 0)
+				return
+			}
+			if (resultCode == Activity.RESULT_OK) {
+				val selectedPaths: MutableList<String> = mutableListOf()
+				if (data?.clipData != null) {
+					// Handle multiple file selection
+					val clipData = data.clipData
```

Comment:
Do this before the check for `null`, this will allow you to remove the `!!` below.